x/umee/keeper/keeper.go

```diff
@@ -29,3 +31,141 @@ func NewKeeper(cdc codec.Codec, storeKey, memKey sdk.StoreKey) *Keeper {
 func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
+
+// IsAcceptedAsset returns true if a given denom is an accepted asset (not uToken) type.
+func (k Keeper) IsAcceptedAsset(ctx sdk.Context, denom string) bool {
+	if k.GetAssociatedUtoken(ctx, denom) != "" {
+		// Has an associated utoken if (and only if) it's an accepted asset
+		return true
+	}
+	return false
+}
+
+// IsAcceptedUtoken returns true if a given denom is an accepted uToken (not asset) type.
```

Comment:
Is this just for borrowing?

Reply:
Might be for lending too, unless we want people to be able to lend tokens we haven't whitelisted (and even then, no one could borrow them) - so I think the accepted Asset+Utoken lists are shared between borrowing and lending (and whenever an Asset is whitelisted, its associated uToken is whitelisted in the same action.